src/utils.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * @license Copyright (c) 2003-2016, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+import { widgetize, isWidget } from './widget/utils.js';
+
+const imageSymbol = Symbol( 'isImage' );
+
+/**
+ * Converts given {@link engine.view.Element} to image widget:
+ * * adds {@link engine.view.Element#addCustomProperty custom property} allowing to recognize image widget element,
+ * * calls {@link image.widget.utils.widgetize widgetize}.
+ *
+ * @param {engine.view.Element} viewElement
+ * @returns {engine.view.Element}
+ */
+export function toImageWidget( viewElement ) {
```

Comment:
I think I really like this approach. No subclassing. Everything happens by extension (through custom props). And the exposed API is `make()` and `check()`.